### PR TITLE
feat(icons): social icon set added to core

### PR DESCRIPTION
**Pull request by mathisscott**
Signed-off-by: Scott Mathis <smathis@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [x] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #3911 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/clr-core/icon-shapes/collections/social.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  [calendarIconName, ['date']],
+];
+
+/**
+ * Function that can be called to load the core icon set.
+ *
+ * ```typescript
+ * import { loadSocialIconSet } from '@clr/core/icon';
+ *
+ * loadSocialIconSet();
```
Comment (coryrylan) [inline src/clr-core/icon-shapes/collections/social.ts]: non blocking nit: update tsdoc example

```typescript
 import '@clr/core/icon';
 import { loadTravelIconSet } from '@clr/core/icon-set';

 loadTravelIconSet();
```

File: src/clr-core/icon-shapes/collections/social.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    [thumbsDownIconName]?: string;
+    [thumbsUpIconName]?: string;
+  }
+}
+
+/*
+
+Object.defineProperty(SocialShapes, 'favorite', descriptorConfig(SocialShapes.star));
+Object.defineProperty(SocialShapes, 'email', descriptorConfig(SocialShapes.envelope));
+Object.defineProperty(SocialShapes, 'date', descriptorConfig(SocialShapes.calendar));
```
Comment (coryrylan) [inline src/clr-core/icon-shapes/collections/social.ts]: Dead code?

Comment (mathisscott) [reply src/clr-core/icon-shapes/collections/social.ts]: Yep. Will remove.

File: src/clr-core/icon-shapes/collections/social.ts (hunk 1 above)
Comment (mathisscott) [reply src/clr-core/icon-shapes/collections/social.ts]: Will do. Thanks!

**Pull request merged**